src/features/storage/index.ts

```diff
@@ -1,10 +1,32 @@
-import { HostnameStorage } from "../../constant";
+import { HostnameStorage, SyncSupportStorage } from "../../constant";
+import { decodesyncSupport } from "../setting/decode"
+
+export const fromStorage = async <T>(
+    hostname: string,
+    key: string,
+    decoder: (data: any) => T,
+    allowSync = true
+): Promise<T> => {
+    const storageKey = hostname + "-" + key;
+    if (allowSync) {
+        const syncSupport = await fromStorage<boolean>(hostname, SyncSupportStorage, decodesyncSupport, false);
+        if (syncSupport) {
+            return new Promise(function (resolve) {
+                chrome.storage.sync.get(storageKey, function (items: any) {
+                    if (storageKey in items) {
+                        resolve(decoder(items[storageKey]));
+                    } else {
+                        resolve(decoder(undefined));
+                    }
+                });
+            });
+        }
+    }
 
-export const fromStorage = <T>(hostname: string, key: string, decoder: (data: any) => T): Promise<T> => {
     return new Promise(function (resolve) {
-        chrome.storage.local.get(hostname, function (items: any) {
-            if (hostname in items && key in items[hostname]) {
-                resolve(decoder(items[hostname][key]));
+        chrome.storage.local.get(storageKey, function (items: any) {
+            if (storageKey in items) {
+                resolve(decoder(items[storageKey]));
```

Comment:
既存の状態（syncがOFF）では`hostname`をkeyにして保存しているので，`storageKey`に変えてしまうとアプデでメモやチェックボックス，設定などが消えてしまうので今のままの方がいいかもです
（何か理由があって変更していたのであれば教えてください〜🙌）
```suggestion
        chrome.storage.local.get(hostname, function (items: any) {
            if (hostname in items && key in items[hostname]) {
                resolve(decoder(items[hostname][key]));
```

Reply:
storage.syncのところだけkeyを変えるつもりだったのですが、storage.localまで変えてしまっていたみたいです。
修正ありがとうございます。